### ISSUE TITLE
consider merging/converging with SyncIt

### ISSUE THREAD

**Issue by michielbdejong** [remotestorage/remotestorage.js#787]
Looking at https://github.com/forbesmyester/SyncIt it looks like they built exactly what we built with what was introduced as "the sync lib" and then became part of remotestorage.js starting with version 0.6. They will have a different set of supported backends, a different app-facing API.

Probably, in some things, our sync will be better, in others, theirs will be. In any case, it would make sense to compare notes.

Say we add a remoteStorage backend to SyncIt and the result is better than what we have now, then we don't really need to continue developing the sync part of our library separately. We could use SyncIt as our sync engine, which would come with all backends supported, get our discovery from webfinger.js, and then all we need to maintain ourselves is basically the widget (and the modules repo on top of the whole thing, of course).

On the other hand, if we add a remoteStorage backend to SyncIt, and the result is worse than what we have in-house, then we can send them a bunch of pull requests to get them up to date with stuff like multi-thread async sync and IndexedDB commit cache. Then once those land, we could still switch to SyncIt.

Unfortunately, I don't have a lot of time the coming six months, but at least I think it's good if we have a ticket open about this. :)

cc @forbesmyester

**Comment by michielbdejong** [comment]
Instead of trying to merge remotestorage.js and SyncIt, I will play around with combining localForage + SyncIt + ludbud.

**Comment by forbesmyester** [comment]
Hi Michiel,

I always intended to do the localForage + SyncIt, I'm actually confident I
can do that super quickly. Would that help? I'll have to look into ludbud
and what it does...

Thanks

Matt

**Comment by michielbdejong** [comment]
@forbesmyester OK, so this project, remotestorage.js consists more or less of the following parts:

* UI widget
* helper functions for dealing with domain modules
* local store (we wrote our own one, but it's equivalent to localForage)
* sync engine (this part is equivalent to SyncIt, as I understand)
* it uses webfinger.js for discovery
* simple OAuth client for obtaining the access token
* what we call the "wire client" (doing the actual calls to the server's API)
* experimental wire clients for Dropbox and Google Drive support.

Now, ludbud does just webfinger+oauth+wireclient, in 300 lines of code.

It's a light-weight client for the remoteStorage protocol that does no sync etc., it only does the actual requests to the server. You can see an example on https://ludbud.5apps.com/.

**Comment by forbesmyester** [comment]
So does localForage + SyncIt + ludbud possibly replace everything except
the UI components?

**Comment by michielbdejong** [comment]
Pretty much, yes. UI components and also https://github.com/remotestorage/modules would still be missing. I'll try to build it! :)

**Comment by forbesmyester** [comment]
Hi All,

I've made progress on this... see the original bug we had ( https://github.com/forbesmyester/SyncIt/issues/5 ) for details. Basically I have the LocalForage version passing mostly all the relevant tests in SyncIt. I'm pretty happy it's fully compatible with the old LocalStorage based on but I'd like to get my TodoMVC app using it just to be sure.

The code is at https://github.com/forbesmyester/SyncIt/tree/localforage

**Comment by forbesmyester** [comment]
LocalForage branch merged into Master. TodoMVC project hosted at https://github.com/forbesmyester/SyncItBootstrap/tree/todomvc ported to use LocalForage (and it works fine). So SyncIt backed by LocalForage is good to go... see https://github.com/forbesmyester/SyncItBootstrap/blob/master/public/js/SyncItFactory.js#L63 for usage.

**Comment by michielbdejong** [comment]
\o/ will check it out soon! I've been working on https://github.com/Decentralized-Sharing-Working-Group/server-to-server lately. Announced it on the "unhosted web apps" Google Group though, and tweeted about it! :)
